src/libs/ajax/analysis-providers/AnalysisProvider.ts

```diff
@@ -0,0 +1,48 @@
+import { Ajax } from 'src/libs/ajax'
+import { GoogleWorkspaceInfo, isGoogleWorkspaceInfo, WorkspaceInfo } from 'src/libs/workspace-utils'
+import { AnalysisFile } from 'src/pages/workspaces/workspace/analysis/useAnalysisFiles'
+import { getExtension, stripExtension } from 'src/pages/workspaces/workspace/analysis/utils/file-utils'
+import { ToolLabel } from 'src/pages/workspaces/workspace/analysis/utils/tool-utils'
+
+
+export interface AnalysisProviderContract {
```

Comment:
Hmm... this seems a bit in conflict with `useAnalysisFiles`. The `list` is captured there already, perhaps just add copy to it?

Reply:
I agree it seems like things are duplicated.
However, I think the long term goal is to move to the Provider pattern? It seems like useAnalysisFiles is just that just with a different name.
I would like Miguel's input, though.

Reply:
ok, yes, there is overlap now that useAnalysisFiles has become a thing.  Looking at useAnalysisFiles though, and it seems to be tied to a specific assumption on it being consumed in a certain form.  I think it makes sense for it to keep that purpose, but to delegate the "is azure or GCP" related call logic to the provider being introduced here.  Add the needed verbs to this provider here so that useAnalysisFiles can delegate the cloud platform awareness to this new provider and reduce it's own complexity a bit.

Reply:
So, something like this? Commented lines are the old code in useAnalysisFiles.ts

```
const refresh = withHandlers([
    withErrorReporting('Error loading analysis files'),
    Utils.withBusyState(setLoading)
  ], async (): Promise<void> => {
    await AnalysisProvider.listAnalyses(workspace.workspace, signal).then(setAnalyses)
    // const workspaceInfo = workspace.workspace
    // const existingAnalyses: AnalysisFile[] = isGoogleWorkspaceInfo(workspaceInfo) ?
    //   await Ajax(signal).Buckets.listAnalyses(workspaceInfo.googleProject, workspaceInfo.bucketName) :
    //   await Ajax(signal).AzureStorage.listNotebooks(workspaceInfo.workspaceId)
    // setAnalyses(existingAnalyses)
  })
  ```

Reply:
close, but don't use .then with async await as we discussed in slack.  The other verbs in useAnalysisFiles would similarly be directed to new implementations of those verbs as needed in AnalysisProvider.ts

Reply:
Yes! The changes should be reflected now.

I'm also wondering about the naming in useAnalysisFiles - do we want it to mirror the function naming in AnalysisProvider?
Personally, I don't like the plain "create" because it's really vague. But I'm not sure that we should have them both named "createAnalysis" either.
Any thoughts?

Reply:
the createAnalysis/deleteAnalysis naming I see now over in useAnalysisFiles seems reasonable.